src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -69,9 +70,14 @@ impl<S: Append + 'static> Coordinator<S> {
         }
 
         let isolation_level = session.vars().transaction_isolation();
-        let use_timestamp_oracle = isolation_level == &vars::IsolationLevel::StrictSerializable
-            && timeline.is_some()
-            && when.advance_to_global_ts();
+        // We only care about timelines for Strict Serializable and non AS OF queries.
+        let timeline = if isolation_level == &vars::IsolationLevel::StrictSerializable
+            && when.advance_to_global_ts()
+        {
+            timeline_context.timeline()
```

Comment:
Why is it ok for `TimestampDependent` queries to return `None` here, and thus not trigger the `if` block below that does advance to oracle read ts?